Sources/PrimerSDK/Classes/Core/Payment Services/ApayaService.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  ApayaService.swift
+//  PrimerSDK
+//
+//  Created by Carl Eriksson on 26/07/2021.
+//
+
+#if canImport(UIKit)
+
+internal protocol ApayaServiceProtocol {
```

Comment:
I've noticed that all oauth payment methods (at least until now) have a way to create a payment session. I think that the abstraction should go at this level. Creating a protocol for each service, doesn't abstract anything.

Implementation could be sth like:
```
protocol OAuthServiceProtocol {
    func createPaymentSession(_ completion: @escaping (Result<String, Error>) -> Void)
}

// Then if any of the oauth services have any specifics

protocol ApayaServiceProtocol: OAuthServiceProtocol {
    func doSth()
}

internal class ApayaService: ApayaServiceProtocol {

    func createPaymentSession(_ completion: @escaping (Result<String, Error>) -> Void) {

    }

    func doSth() {

    }
}

// Same goes for the PrimerAPIClientProtocol function apayaCreateSession
// It could be createOAuthSession and get the url as a param.

```

Reply:
great idea! will try it out